R/counts_labels.R

```diff
@@ -0,0 +1,39 @@
+#' Counts the number of observations in each category of a variable that has
+#' value labels, returns counts and percent of total.
+#' @export
+#' @title counts_labels
+#' @param data a data.frame
+#' @param x a column in 'data'. Must be a string.
+#' @import dplyr
+counts_labels <- function(data, x) {
+  if(is.factor(data[[x]])) {
+    data[[x]] <- hablar::convert_num(data[[x]])
+    cat("Factor converted to numeric.")
+  }
+  if(is.character(data[[x]])) {
+    data[[x]] <- hablar::convert_num(data[[x]])
+    cat("Factor converted to numeric.")
+  }
+  if(is.null(sjlabelled::get_labels(data[[x]]))) {
+    stop(paste0("'", x, "' does not have value labels.")) 
+  }
+  dims <- dim(data)
+  counts <- data %>%
+    select_(x) %>%
+    group_by_(x) %>%
+    count_()
+  labels <- tibble::tibble(variable_label = 
+                     get_labels(data[[x]]))
+  data.table::setDT(labels, keep.rownames = TRUE)[]
```

Comment:
Seems weird to me that you are mixing and matching `data.table()` functionality with `dplyr` functionality. Is there a reason `data.table()` is necessary?